python/lsst/sims/utils/timeTransformations.py

```diff
@@ -0,0 +1,195 @@
+
+    if utc<Ut1MinusUtcData._mjd_arr[0] or utc>Ut1MinusUtcData._mjd_arr[-1]:
+        msg = "UTC = %e is outside of the the bounds " % utc \
+               + "for which we have UT1-UTC "\
+               + "data (%e <= utc <= %e)\n" % (Ut1MinusUtcData._mjd_arr[0], Ut1MinusUtcData._mjd_arr[-1]) \
+               + "We will return UT1-UTC = 0, for lack of a better idea"
+        warnings.warn(msg)
+
+        return 0.0
+
+    return np.interp(utc, Ut1MinusUtcData._mjd_arr, Ut1MinusUtcData._dut_arr)
+
+
+def ut1FromUtc(utc):
+    """
+    Use data downloaded from
+
+    ftp://cddis.gsfc.nasa.gov/pub/products/iers/
+
+    to transform UTC into UT1
+
+    @param [in] UTC as an MJD
+
+    @param [out] UT1 as an MJD
+    """
+
+    sec_to_days = 1.0/86400.0
+    dt = dutFromUtc(utc)
+    return utc + dt*sec_to_days
+
+
+def utcFromUt1(ut1):
+    """
+   Use data downloaded from
+
+    ftp://cddis.gsfc.nasa.gov/pub/products/iers/
+
+    to transform UT1 into UTC
+
+    @param [in] UT1 as an MJD
+
+    @param [out] UTC as an MJD
+
+    Note: because we only have data for UT1-UTC as a function
+    of UTC, this method operates by creating arrays of UT1 and UTC
+    from ut1FromUtc() and interpolating them.
+    """
+
+    utc_arr = np.arange(ut1-1.0, ut1+1.0, 0.25)
+    ut1_arr = np.array([ut1FromUtc(utc) for utc in utc_arr])
+    return np.interp(ut1, ut1_arr, utc_arr)
+
+
+def dttFromUtc(utc):
+    """
+    Use data downloaded from
+
+    ftp://tai.bipm.org/TFG/TT(BIPM)/TTBIPM.14
+
+    to to calculate TT-TAI in seconds as a function of UTC
+
+    @param [in] UTC as an MJD
+
+    @param [out] TT-TAI in seconds
+
+    Note: The bounds of our data are 42589.0 < UTC < 57019.0
+
+    for UTC < 42589.0 this method returns 32.184 seconds
+
+    for UTC > 57019.0 this method returns 32.184 seconds + 27.697 microseconds
+
+    Discussion of the precision of this method can be found at the URL above.
+    Roughly speaking, the precision is 1 ns before UTC=52729; 0.1 ns after that.
+    """
+
+    if utc>57019.0:
+        dt = 27.6970
+    elif utc<42589.0:
+        dt = 0.0
+    else:
+        dt = np.interp(utc, TT_from_TAI_Data._mjd_arr, TT_from_TAI_Data._dt_arr)
+
+    return 32.1840 + dt*1.0e-6
+
+
+def ttFromTai(tai):
+    """
+    Return Terestrial Time (TT) as a function of TAI.
+
+    Because of numerical precision, this method assumes
+
+    TT = TAI + 32.184 seconds
+
+    for better precision, find TT-TAI using the method dttFromUtc
+    """
+    return tai + 0.00037250
+
+
+def tdbFromTt(tt):
+    """
+    Return TDB (Barycentric Dynamical Time) from TT (Terrestrial Time)
+    using equation 2.222-1 from
+
+    Seidelmann 1992, "Explanatory Supplement to the Astronomical Almanac"
+    Unviversity Science Books
+
+    @param [in] tt as an MJD
+
+    @param [out] TDB as an MJD
+
+    Note: this method should only be accurate to within a few tens of
+    microseconds (because of General Relativistic corrections that have
+    been neglected)
+    """
```

Comment:
So, TDB is pretty commonly used in orbits and ephemeris calculations. 
It may be that we don't care for catalogs and maf, because the TDB/TT/TAI conversion for the orbital information when predicting positions is also taken care of within Oorb. 
The erfa library suggests you can calculate your own TT-TBD difference using pre-calculated tables  (https://github.com/liberfa/erfa/blob/master/src/tttdb.c) or gives you the option of using a calculation that seems much bigger than this snippet : https://github.com/liberfa/erfa/blob/master/src/dtdb.c 
Is the comment "General Relativisitic Corrections that have been neglected" completely correct? Because the TT-TBD difference is mostly (entirely?) due to relativistic corrections, so it sounds very worrying to hear that they've been neglected. I could be misinterpreting this.